dashboard/app/views/shared/_courseblock_action.haml

```diff
@@ -1,12 +1,16 @@
 - if current_user && showState
   .clear
+
   - script = Script.find_by_name(id)
-  - if current_user.completed_scripts.include?(script)
+  - user_script = current_user.user_scripts.where(script_id: script.id).first
+  - next_unpassed_progression_level = current_user.next_unpassed_progression_level(script)
+
+  - if user_script.try(:completed_at) || (user_script && next_unpassed_progression_level.nil?)
```

Comment:
good call @laurelfan .. i think this does the trick and makes sure that even the users with wonky state see their 'print certificate' option too?